File: cpp/bc2534.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
	int qtdN, vNo, nCh, aux;
	
	while (cin >> qtdN >> nCh)
	{
		int vec[qtdN];
		
		for(int i = 0; i < qtdN; i++)
		{
			cin >> vNo;
			vec[i] = vNo;
		}
		
		sort(vec, vec + qtdN, greater<int>());
		
		for(int i = 0; i < nCh; i++)
		{
			cin >> vNo;
			cout << vec[vNo-1] << endl;
		}
		
	}
}

File: python/bc2534.py
while True:
    try:
        line = input().split(" ")

        qtdN = int(line[0])
        nCh = int(line[1])

        vec = [int]*qtdN

        for c in range(0, qtdN):
            vec[c] = int(input())

        vec.sort(reverse=True)

        for c in range(0, nCh):
            val = int(input())
            print(f"{vec[val-1]}")
    except EOFError:
        break